tests/app/views/test_agreements.py

```diff
@@ -1,23 +1,156 @@
 import json
 import pytest
 from datetime import datetime
-from app.models import AuditEvent, db, FrameworkAgreement
+from freezegun import freeze_time
+from app.models import AuditEvent, db, Framework, FrameworkAgreement
 from ..helpers import BaseApplicationTest, fixture_params
 
 
-class TestGetFrameworkAgreement(BaseApplicationTest):
-
+class BaseFrameworkAgreementTest(BaseApplicationTest):
     def create_agreement(self, supplier_framework, **framework_agreement_kwargs):
         with self.app.app_context():
+            framework = Framework.query.filter(Framework.slug == supplier_framework['frameworkSlug']).first()
+
             agreement = FrameworkAgreement(
-                supplier_id=supplier_framework['supplier_id'],
-                framework_id=supplier_framework['framework_id'],
+                supplier_id=supplier_framework['supplierId'],
+                framework_id=framework.id,
                 **framework_agreement_kwargs)
             db.session.add(agreement)
             db.session.commit()
 
             return agreement.id
 
+
+class TestCreateFrameworkAgreement(BaseApplicationTest):
+    def post_create_agreement(self, supplier_id=None, framework_slug=None):
+        agreement_data = {}
+        if supplier_id:
+            agreement_data['supplierId'] = supplier_id
+        if framework_slug:
+            agreement_data['frameworkSlug'] = framework_slug
+
+        return self.client.post(
+            '/agreements',
+            data=json.dumps(
+                {
+                    'updated_by': 'interested@example.com',
+                    'agreement': agreement_data
+                }),
+            content_type='application/json')
+
+    def test_can_create_framework_agreement(self, supplier_framework):
+        res = self.post_create_agreement(
+            supplier_id=supplier_framework['supplierId'],
```

Comment:
This is the magic of fixtures, @TheDoubleK!

Once you define a fixture (ie, [`supplier_framework`](https://github.com/alphagov/digitalmarketplace-api/blob/master/tests/conftest.py#L41)), then you can simply pass it into any test as an argument. 

[There's a simple example on the first page of the docs](http://doc.pytest.org/en/latest/fixture.html#fixtures-as-function-arguments).

✨ 🔮 ✨